src/cpp/shared_core/FilePath.cpp

```diff
@@ -319,16 +319,32 @@ FilePath::FilePath() :
 {
 }
 
+FilePath::FilePath(const char* in_absolutePath)
+{
+   if (nullptr == in_absolutePath)
+      m_impl.reset(new Impl());
```

Comment:
One thing that scares me a bit (that I hadn't considered before): if we're getting something that's accidentally / unintentionally a null pointer, could we end up forming paths to files we didn't intend to? E.g. if we had something like:

```
const char* path = foo()  // oops, gave a null pointer
FilePath(path).completePath("log").ensureDirectory();
```

What will happen in this case? It feels like we should log an error and throw an exception instead (or otherwise just abort).

Reply:
That's a good point. I _believe_ `getAbsolutePath()` would give `/current/working/dir/log` if called on the completed path in your example.

Reply:
If we do throw an exception though, we'll most likely end up in the same situation as before. Like @kfeinauer mentioned, we'd be trading one crash for another. But perhaps a more  informative crash is still an improvement?